worlds/kh1/__init__.py

```diff
@@ -0,0 +1,314 @@
+            self.multiworld.get_location("Hollow Bastion Entrance Hall Emblem Piece (Flame)", self.player).place_locked_item(self.create_item("Emblem Piece (Flame)"))
+            self.multiworld.get_location("Hollow Bastion Entrance Hall Emblem Piece (Statue)", self.player).place_locked_item(self.create_item("Emblem Piece (Statue)"))
+            self.multiworld.get_location("Hollow Bastion Entrance Hall Emblem Piece (Fountain)", self.player).place_locked_item(self.create_item("Emblem Piece (Fountain)"))
+            self.multiworld.get_location("Hollow Bastion Entrance Hall Emblem Piece (Chest)", self.player).place_locked_item(self.create_item("Emblem Piece (Chest)"))
+
+    def get_filler_item_name(self) -> str:
+        fillers = {}
+        disclude = []
+        fillers.update(get_items_by_category("Item", disclude))
+        fillers.update(get_items_by_category("Camping", disclude))
+        fillers.update(get_items_by_category("Stat Ups", disclude))
+        weights = [data.weight for data in fillers.values()]
+        return self.random.choices([filler for filler in fillers.keys()], weights, k=1)[0]
+
+    def fill_slot_data(self) -> dict:
+        slot_data = {"xpmult": int(self.options.exp_multiplier)/16,
+                    "required_reports_eotw": self.determine_reports_required_to_open_end_of_the_world(),
+                    "required_reports_door": self.determine_reports_required_to_open_final_rest_door(),
+                    "door": self.options.final_rest_door.current_key,
+                    "seed": self.multiworld.seed_name,
+                    "advanced_logic": bool(self.options.advanced_logic),
+                    "hundred_acre_wood": bool(self.options.hundred_acre_wood),
+                    "atlantica": bool(self.options.atlantica),
+                    "goal": str(self.options.goal.current_key)}
+        if self.options.randomize_keyblade_stats:
+            min_str_bonus = min(self.options.keyblade_min_str, self.options.keyblade_max_str)
+            max_str_bonus = max(self.options.keyblade_min_str, self.options.keyblade_max_str)
+            min_mp_bonus = min(self.options.keyblade_min_mp, self.options.keyblade_max_mp)
+            max_mp_bonus = max(self.options.keyblade_min_mp, self.options.keyblade_max_mp)
+            slot_data["keyblade_stats"] = ""
+            for i in range(22):
+                if i < 4 and self.options.bad_starting_weapons:
+                    slot_data["keyblade_stats"] = slot_data["keyblade_stats"] + "1,0,"
+                else:
+                    if min_str_bonus != max_str_bonus:
+                        str_bonus = int(self.random.randrange(min_str_bonus,max_str_bonus))
+                    else:
+                        str_bonus = int(min_str_bonus)
+                    if min_mp_bonus != max_mp_bonus:
+                        mp_bonus = int(self.random.randrange(min_mp_bonus,max_mp_bonus))
+                    else:
+                        mp_bonus = int(min_mp_bonus)
```

Comment:
So this actually _excludes_ the max, because randrange is max-exclusive, you can just do this instead
```suggestion
                    str_bonus = int(self.random.randint(min_str_bonus, max_str_bonus))
                    mp_bonus = int(self.random.randint(min_mp_bonus, max_mp_bonus))
```